pkg/metrics/workqueue.go

```diff
@@ -87,14 +87,18 @@ var (
 	}, []string{"name"})
 )
 
+var WorkqueueRegistry = prometheus.NewRegistry()
+
 func init() {
-	Registry.MustRegister(depth)
-	Registry.MustRegister(adds)
-	Registry.MustRegister(latency)
-	Registry.MustRegister(workDuration)
-	Registry.MustRegister(unfinished)
-	Registry.MustRegister(longestRunningProcessor)
-	Registry.MustRegister(retries)
+	WorkqueueRegistry.MustRegister(depth)
+	WorkqueueRegistry.MustRegister(adds)
+	WorkqueueRegistry.MustRegister(latency)
+	WorkqueueRegistry.MustRegister(workDuration)
+	WorkqueueRegistry.MustRegister(unfinished)
+	WorkqueueRegistry.MustRegister(longestRunningProcessor)
+	WorkqueueRegistry.MustRegister(retries)
+
+	Registry.Register(WorkqueueRegistry)
```

Comment:
I understand the problem, but this means that the workqueue metrics will disappear for all existing users. That is not an option. We need to find a solution that does not break existing users.

/hold

Reply:
Hey @alvaroaleman , these metrics won't disappear actually. We are adding those metrics back to the main `Registry` with `Registry.Register(WorkqueueRegistry)`. So this won't affect anything for existing users.